### ISSUE TITLE
How to write to bvh file?

### ISSUE THREAD

**Issue by mariolew** [akanazawa/motion_reconstruction#8]
Hi, I noticed that '# from jason.bvh_core import write2bvh' this line has been commented out, so how to write to bvh?

**Comment by sxhzju** [comment]
It's not implenmented yet but I have implenmented myself.
How about exchange? You give me ASI, I give you that.

**Comment by mariolew** [comment]
@Zju-George I've converted the result motion already so that I can use it to train policy, I have obtained the root translation myself, but I think it's not good enough, now I want to know how the author compute the root translation.

**Comment by sxhzju** [comment]
I have also obtained The root translation using optimal method, Maybe WE can compare The result using The same video clip? Also, did you implement ASI? And does it improve a lot?

**Comment by mariolew** [comment]
Yes, I've implemented ASI, it converges much faster than RSI, it obtains good results on some easy clips(those shows in the paper), but it still fails if the motion result is not good enough.  Maybe you can show me some result on caixukun?

**Comment by sxhzju** [comment]
I am amazed that you know caixukun! Hahaha, but I wanna share you something more difficult. Actually when camera is not moving or rotating the method is robust but I have also figured out that when camera has only rotation, I can also obtain the root trans.

![test](https://user-images.githubusercontent.com/33959089/64143944-ed8c3e80-ce44-11e9-9886-24f256a5cd6a.gif)
![Picture2](https://user-images.githubusercontent.com/33959089/64143954-f9780080-ce44-11e9-84ae-18ef87fc5494.png)

**Comment by sxhzju** [comment]
But I also encounter one big problem, we have no prior limitation on the ankle joints but when ankle joints are not touching the ground or have some wierd rotation, it is very hard to train the policy. And I think it is something that cannot be handled by ASI, so do you have any idea about this?

**Comment by ManifoldFR** [comment]
Hi, we are trying to reproduce results from the paper and are stuck at the export phase between SFV and DeepMimic. We have a working [BVH to DeepMimic translator](https://github.com/BartMoyaers/BvhToDeepMimic) but are not sure about the format of the BVH file. There's a line in `refine_video.py` that invokes a function `write2bvh` but it's not included or mentioned anywhere else.

Thank you for your help.

**Comment by lmontaut** [comment]
Hi @Zju-George, I am wondering how you calculated the global root translation you mentioned above ? Can you explain how you did that ? Thank you very much !

**Comment by sxhzju** [comment]
> Hi @Zju-George, I am wondering how you calculated the global root translation you mentioned above ? Can you explain how you did that ? Thank you very much !

By estimating the camera movement first ( I use OpenCV, but of course you can use any camera tracking software like PFTrack), then I optimize the 2D joints loss using the camera information.

**Comment by sxhzju** [comment]
> Hi, we are trying to reproduce results from the paper and are stuck at the export phase between SFV and DeepMimic. We have a working [BVH to DeepMimic translator](https://github.com/BartMoyaers/BvhToDeepMimic) but are not sure about the format of the BVH file. There's a line in `refine_video.py` that invokes a function `write2bvh` but it's not included or mentioned anywhere else.
> 
> Thank you for your help.

Check this Zju-George/DeepMimic#2 out！

**Comment by hellozjj** [comment]
> I've converted the result motion already so that I can use it to train policy, I have obtained the root translation myself, but I think it's not good enough, now I want to know how the author compute the root translation.

Hi, @mariolew   how do you obtain the root translation?  I couldn't figure out how to estimate the root position. Can you give any clue? Thanks very much!
